contracts/pcv/PCVGuardian.sol

```diff
@@ -0,0 +1,204 @@
+
+    /// @notice governor-or-guardian-only method to remove pcvDeposit address from the whitelist to withdraw funds from
+    /// @param pcvDeposit the address to remove from whitelist
+    function removeWhitelistAddress(address pcvDeposit)
+        external
+        override
+        onlyGuardianOrGovernor
+    {
+        _removeWhitelistAddress(pcvDeposit);
+    }
+
+    /// @notice batch version of removeWhitelistAddress
+    /// @param _whitelistAddresses the addresses to remove from whitelist
+    function removeWhitelistAddresses(address[] calldata _whitelistAddresses)
+        external
+        override
+        onlyGuardianOrGovernor
+    {
+        // improbable to ever overflow
+        unchecked {
+            require(_whitelistAddresses.length != 0, "empty");
+            for (uint256 i = 0; i < _whitelistAddresses.length; i++) {
+                _removeWhitelistAddress(_whitelistAddresses[i]);
+            }
+        }
+    }
+
+    /// @notice governor-or-guardian-or-pcv-guard method to withdraw funds from a pcv deposit, by calling the withdraw() method on it
+    /// @param pcvDeposit the address of the pcv deposit contract
+    /// @param amount the amount to withdraw
+    function withdrawToSafeAddress(address pcvDeposit, uint256 amount)
+        external
+        override
+        hasAnyOfThreeRoles(
+            TribeRoles.GOVERNOR,
+            TribeRoles.GUARDIAN,
+            TribeRoles.PCV_GUARD
+        )
+        onlyWhitelist(pcvDeposit)
+    {
+        _withdrawToSafeAddress(pcvDeposit, amount);
+    }
+
+    /// @notice governor-or-guardian-or-pcv-guard method to withdraw all at once funds from a pcv deposit, by calling the withdraw() method on it
+    /// @param pcvDeposit the address of the pcv deposit contract
+    function withdrawAllToSafeAddress(address pcvDeposit)
+        external
+        override
+        hasAnyOfThreeRoles(
+            TribeRoles.GOVERNOR,
+            TribeRoles.GUARDIAN,
+            TribeRoles.PCV_GUARD
+        )
+        onlyWhitelist(pcvDeposit)
+    {
+        _withdrawToSafeAddress(pcvDeposit, IPCVDeposit(pcvDeposit).balance());
+    }
+
+    /// @notice governor-or-guardian-only method to withdraw an ERC20 from a pcv deposit, by calling the withdrawERC20() method on it
+    /// @param pcvDeposit the deposit to pull funds from
+    /// @param token the address of the token to withdraw
+    /// @param amount the amount of funds to withdraw
+    function withdrawERC20ToSafeAddress(
+        address pcvDeposit,
+        address token,
+        uint256 amount
+    )
+        external
+        override
+        hasAnyOfThreeRoles(
+            TribeRoles.GOVERNOR,
+            TribeRoles.GUARDIAN,
+            TribeRoles.PCV_GUARD
+        )
+        onlyWhitelist(pcvDeposit)
+    {
+        IPCVDeposit(pcvDeposit).withdrawERC20(token, safeAddress, amount);
+
+        emit PCVGuardianERC20Withdrawal(pcvDeposit, token, amount);
```

Comment:
Let's package this logic up into an internal helper method and leverage it for the `withdrawAllERC20ToSafeAddress` function.